source/extensions/transport_sockets/tcp_stats/tcp_stats.cc

```diff
@@ -145,14 +152,72 @@ void TcpStatsSocket::recordStats() {
                  tcp_info->tcpi_bytes_received);
   update_counter(config_->stats_.cx_tx_bytes_sent_, last_cx_tx_bytes_sent_,
                  tcp_info->tcpi_bytes_sent);
+  update_counter(config_->stats_.cx_data_segments_delivered_, last_cx_data_segments_delivered_,
+                 tcp_info->tcpi_delivered);
+  update_counter(config_->stats_.cx_reordering_, last_cx_reordering_, tcp_info->tcpi_reordering);
 
   update_gauge(config_->stats_.cx_tx_unsent_bytes_, last_cx_tx_unsent_bytes_,
                tcp_info->tcpi_notsent_bytes);
   update_gauge(config_->stats_.cx_tx_unacked_segments_, last_cx_tx_unacked_segments_,
                tcp_info->tcpi_unacked);
+  update_gauge(config_->stats_.cx_rto_us_, last_cx_rto_us_, tcp_info->tcpi_rto);
```

Comment:
All of these that are time values don't make sense to add/combine. Remove them or consider making them histograms.